CMakeLists.txt

```
@@ -104,10 +104,20 @@ install(
 )
 
 install(
-  DIRECTORY ${_ouster_ros_INCLUDE_DIRS}
+  DIRECTORY include/ouster_ros/
   DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
 )
 
+install(
+  DIRECTORY ouster-sdk/ouster_client/include/ouster/
+  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}/../ouster
+)
+
+install(
+  DIRECTORY ouster-sdk/ouster_client/include/optional-lite/nonstd/
+  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}/../nonstd
+)
+
```

> This makes sense but only concern is there were different dependencies (ROS packages) using the nonstd::optional library and they by default did the same thing, wouldn't they collide when installing to the same folder.

I see your point. Do you think it would be okay to just not install it? We are primarily interested in the Ouster headers here.
```suggestion
```

---

How about we just place it in the same directory under ouster? this way it is in the same folder as `types.h` which makes use of this header file and at the same we don't run into collision, so this would be my proposal:
```suggestion
install(
  DIRECTORY ouster-sdk/ouster_client/include/optional-lite/nonstd/
  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}/../ouster/nonstd/
)
```